Automattic/jetpack
# Unload masterbar and nav unification when nav redesign is enabled

Fixes #

## Proposed changes:

* Updates the action bar module to not load masterbar and nav unification based off a filter which we can override from other areas such as wpcomsh/jetpack-mu. The default value for this filter checks both if the user is proxied and if they have the 'classic' interface setting enabled.

Before - "classic" view:
![Screenshot 2024-02-13 at 11 27 33 AM](https://github.com/Automattic/jetpack/assets/28742426/3d3a1d17-f13e-4d70-9ee5-6d541726387f)

After - "classic" view:
![image](https://github.com/Automattic/jetpack/assets/28742426/b8b8bb37-aafb-4c4b-b34f-f85c0ea91a9d)

NOTE - i forgot to take bigger captures to show the change to nav unification. You can see nav unification is removed in the above image by the comparative width of the sidebar, but you will see it more broadly in testing.  Also note the help center was gone in that image due to its incompatibility with the jetpack-beta-tester plugin, which has been [ fixed here](https://github.com/Automattic/wp-calypso/pull/87430).

Before AND after - "default" view:
![Screenshot 2024-02-13 at 11 26 51 AM](https://github.com/Automattic/jetpack/assets/28742426/72662e34-a055-4cb4-8fdf-3da2c7339ecd)


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Load these changes of jetpack on a WoA site. A bot comment below will give more instructions on using the jetpack beta testing plugin.
* Go to hosting config and select "classic" view, make sure you are proxied.
* visit wp-admin, there should be no masterbar or nav unification customizations.
* disable the proxy - previous customizations should reappear.
* re-enable the proxy, go back to hosting config and toggle the "default" setting.
* back to wp-admin, verify there is no change from previous behvario.
